compiler/rustc_errors/src/diagnostic_impls.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{fluent, DiagnosticBuilder, Handler, IntoDiagnostic};
+use rustc_target::abi::TargetDataLayoutErrors;
+
+impl IntoDiagnostic<'_, !> for TargetDataLayoutErrors<'_> {
+    fn into_diagnostic(self, handler: &Handler) -> DiagnosticBuilder<'_, !> {
```

Comment:
I don't think `rustc_errors` *needs* to depend on `rustc_target`, probably just to implement `IntoDiagnosticArg` on some types. Maybe it would be better to reverse the dependency so that `rustc_target` depends on `rustc_errors`, can use the diagnostic derive to implement this on that type directly, and then can implement `IntoDiagnosticArg` on the whatever we need.

Reply:
We can't because `rustc_errors` depends on `rustc_hir` which depends on `rustc_target`, thus creating the following cyclic package dependency:
 
`rustc_errors` -> `rustc_hir` -> `rustc_target` -> `rustc_errors`.

Then tried to remove `rustc_hir` from `rustc_errors`, but we also can't because we end up with a different cycle: 

`rustc_errors` -> `rustc_lint_defs` -> `rustc_hir` -> `rustc_errors`.